vmware_aria_operations_integration_sdk/mp_build.py

```diff
@@ -108,48 +109,69 @@ def build_subdirectories(directory: str) -> None:
 
 
-def get_registry_components(docker_registry: str) -> Tuple[str, str]:
-    components = docker_registry.split("/")
+def get_registry_components(container_registry: str) -> Tuple[str, str]:
+    components = container_registry.split("/")
     host = components[0]
     path = "/".join(components[1:])
     return host, path
 
 
-def is_valid_registry(docker_registry: str, **kwargs: Any) -> bool:
+def is_valid_registry(container_registry: str, **kwargs: Any) -> bool:
     try:
-        login(docker_registry, **kwargs)
+        if _is_docker_hub_registry_format(container_registry):
+            if "registry_username" not in kwargs:
+                kwargs["registry_username"] = prompt("Enter Docker Hub username: ")
+
+            if "registry_password" not in kwargs:
+                kwargs["registry_password"] = prompt("Password: ", is_password=True)
+            login(**kwargs)
+            container_registry = f"docker.io/{container_registry}"
+
+        login(container_registry=container_registry, **kwargs)
+
     except LoginError:
         return False
 
     return True
 
 
+def _is_docker_hub_registry_format(registry: str) -> bool:
+    # if pattern for dockechub is username/repo
+    pattern = r"^[\w\-]+\/[\w\-]+$"
+
+    return bool(re.match(pattern, registry))
+
+
 def registry_prompt(default: str) -> str:
     return prompt(
-        "Enter the tag for the container registry: ",
+        "Enter the full path for the container registry: ",
         default=default,
         validator=NotEmptyValidator("Host"),
-        description="The tag of a container registry is used to login into the container registry. the tag is composed of\n"
-        "three parts: domain, port, and path. For example:\n"
-        "projects.registry.vmware.com:443/vmware_aria_operations_integration_sdk_mps/base-adapter breaks into\n"
+        description="The path of a container registry is used to login into the container registry. the path is composed of\n"
+        "four parts: domain, port, path, and tag. For example:\n"
+        "projects.registry.vmware.com:443/vmware_aria_operations_integration_sdk_mps/base-adapter:latest breaks into\n"
         "domain: projects.registry.vmware.com\n"
         "port: 443\n"
         "path: vmware_aria_operations_integration_sdk_mps/base-adapter\n"
-        "Port number is optional, and defaults to 443.",
+        "tag: latest\n"
+        "Port number is optional, and defaults to 443.\n"
+        "tag should be omited from the full path"
+        "for Docker Hub repositories simply spevify the path",
     )
 
 
-def get_docker_registry(
+def get_container_registry(
     adapter_kind_key: str,
     config_file: str,
-    docker_registry_arg: Optional[str],
+    container_registry_arg: Optional[str],
     **kwargs: Any,
 ) -> str:
-    docker_registry = get_config_value("docker_registry", config_file=config_file)
+    container_registry = get_config_value("container_registry", config_file=config_file)
```

Comment:
I don't think we should do this without a migration in place, as it will invalidate any existing config files.

Also, we'll need to update anywhere where we reference it in the documentation.

Reply:
I'd rather just write the bit of code that falls back on the old key. Less disruptive and I think less work overall